packages/cozy-client/src/withMutations.jsx

```diff
@@ -24,9 +33,16 @@ const withMutations = (mutations = {}) => WrappedComponent => {
           saveDocument: client.save.bind(client),
           deleteDocument: client.destroy.bind(client)
         },
-        ...(typeof mutations === 'function'
-          ? mutations(client, props)
-          : mutations)
+        ...reduce(
+          mutations.map(
+            mutations =>
+              typeof mutations === 'function'
+                ? mutations(client, props)
+                : mutations
+          ),
+          merge,
+          {}
+        )
```

Comment:
It seems merge can be used directly here.

```
merge(...mutations.map(
            mutations =>
              typeof mutations === 'function'
                ? mutations(client, props)
                : mutations
          ))
```

---

On another level, I wonder if this merging should be the responsibility of the library ? There can be the risk of overwriting :/ It will not be understandable by the user :/

Reply:
I'd be ok with having this in the library. While I haven't had the need to merge mutations, it seems quite legitimate and I wouldn't be surprised to need this as well some day. And as a user, if I specify multiple mutations, it seems expected that mutations with the same name would en up overriding each other in the `props`.